modules/compiler/vecz/source/vecz_pass_builder.cpp

```diff
@@ -181,7 +182,17 @@ bool vecz::buildPassPipeline(ModulePassManager &PM) {
   PM.addPass(createModuleToFunctionPassAdaptor(LowerSwitchPass()));
   PM.addPass(createModuleToFunctionPassAdaptor(FixIrreduciblePass()));
 
-  // We have to run LLVM's Mem2Reg pass in case the front end didn't
+  // It's helpful to run SROA in case it opens up more opportunities to
+  // eliminate aggregates in (particularly SYCL) kernels. This is especially
+  // true after inlining - which we've (usually) just performed in the
+  // BuiltinInliningPass - because otherwise SROA is unable to analyze the
+  // lifetime of allocas due to them being "escaped" by the function callee.
+  PM.addPass(
+      createModuleToFunctionPassAdaptor(SROAPass(SROAOptions::ModifyCFG)));
```

Comment:
This isn't a new issue, but we have a whole bunch of function passes in a row all wrapped in `createModuleToFunctionPassAdaptor`. It would be better to add them all to a single `FunctionPassManager` and wrap that, which would allow analysis results to be reused. Is that something that would make sense to change now, or is that better to leave as a separate PR for later at some point?

Reply:
Oh yeah good point, that's been bugging me for a while. I can open a follow-up PR after this is merged (I'm already half way through testing)?

Reply:
Sounds good to me, thanks.